### PR TITLE
Backport hack/jenkins/e2e.sh to release-1.0

### PR DESCRIPTION
**Pull request by ixdy**
Backport `hack/jenkins/e2e.sh` from right before #17180 was merged, and remove all non-1.0 jobs definitions.

All remaining 1.0 Jenkins jobs (of which there is ... 1?) will use this versioned e2e.sh instead of pulling from master.

Refactoring of upgrade tests is a separate effort.

cc @kubernetes/goog-testing @wonderfly @wojtek-t @piosz
x-ref #17734

### CODE REVIEW

**Comment by wonderfly** [comment]
> All remaining 1.0 Jenkins jobs (of which there is ... 1?) will use this versioned e2e.sh instead of pulling from master.

But non-1.0 jobs will continue pulling from master? Including 1.1 jobs?

**Comment by ihmccreery** [comment]
LGTM.  Thanks @ixdy!

**Comment by ihmccreery** [comment]
> But non-1.0 jobs will continue pulling from master? Including 1.1 jobs?

1.1 jobs will pull from `release-1.1`.  See #17750.

**Comment by ihmccreery** [comment]
(@wonderfly for above comment)

**Comment by ihmccreery** [comment]
Removing LGTM, to wait for @spxtr's verification.

**Comment by ixdy** [comment]
@k8s-bot e2e test this please

**Comment by ixdy** [comment]
@k8s-bot e2e test this please

**Comment by ixdy** [comment]
@k8s-bot e2e test this please

I'm not sure this is ever going to pass. Maybe I'll filter this test if it fails again.

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -60,134 +87,153 @@ if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
...
+
+GCE_RELEASE_SKIP_TESTS=(
+    )
 
 # Define environment variables based on the Jenkins project name.
+# NOTE: Not all jobs are defined here. The hack/jenkins/e2e.sh in master and
+# release branches defines relevant jobs for that particular version of
+# Kubernetes.
 case ${JOB_NAME} in
-  # Runs all non-flaky tests on GCE, sequentially.
```
Comment (wojtek-t) [inline hack/jenkins/e2e.sh]: Do we really want to remove all those suites?

Comment (ihmccreery) [inline hack/jenkins/e2e.sh]: Yup; see https://github.com/kubernetes/kubernetes/issues/17734#issuecomment-159461708.

**Comment by ihmccreery** [comment]
Manually merging.

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -215,18 +266,43 @@ export KUBE_GCE_ZONE=${E2E_ZONE}
 export KUBE_GCE_NETWORK=${E2E_NETWORK}
 export KUBE_GCE_INSTANCE_PREFIX=${KUBE_GCE_INSTANCE_PREFIX:-}
 export KUBE_GCS_STAGING_PATH_SUFFIX=${KUBE_GCS_STAGING_PATH_SUFFIX:-}
+export KUBE_GCE_MINION_PROJECT=${KUBE_GCE_MINION_PROJECT:-}
```
Comment (wonderfly) [inline hack/jenkins/e2e.sh]: I remember myself added these lines for Trusty release jobs, which are now moved to branch 1.1. Don't know if you still want them for 1.0.

File: hack/jenkins/e2e.sh [hunk 3]
```diff
@@ -215,18 +266,43 @@ export KUBE_GCE_ZONE=${E2E_ZONE}
...
+export KUBE_ENABLE_DEPLOYMENTS=${ENABLE_DEPLOYMENTS:-}
+export KUBE_ENABLE_EXPERIMENTAL_API=${ENABLE_EXPERIMENTAL_API:-}
 export MASTER_SIZE=${MASTER_SIZE:-}
 export MINION_SIZE=${MINION_SIZE:-}
+export MINION_DISK_SIZE=${MINION_DISK_SIZE:-}
 export NUM_MINIONS=${NUM_MINIONS:-}
+export TEST_CLUSTER_LOG_LEVEL=${TEST_CLUSTER_LOG_LEVEL:-}
+export TEST_CLUSTER_RESYNC_PERIOD=${TEST_CLUSTER_RESYNC_PERIOD:-}
 export PROJECT=${PROJECT:-}
+export JENKINS_EXPLICIT_VERSION=${JENKINS_EXPLICIT_VERSION:-}
```
Comment (wonderfly) [inline hack/jenkins/e2e.sh]: This too.

File: hack/jenkins/e2e.sh (hunk 2 above)
Comment (ixdy) [inline hack/jenkins/e2e.sh]: I wouldn't worry about it too much. I removed the obvious trusty stuff from the backport, but this basically isn't harming anything.

Comment (wonderfly) [inline hack/jenkins/e2e.sh]: Fair enough.

**Comment by wonderfly** [comment]
@ixdy Just curious though, now that you guys are on 1.1.2 already, why do you keep the release jobs for 1.0 still? Do you release at multiple branches? What's your release cycle like?

**Comment by ihmccreery** [comment]
We expect users to not necessarily upgrade their infrastructure immediately, (either on [GKE](https://cloud.google.com/container-engine/docs/clusters/upgrade) or with other providers,) so our *de facto* release cycle is to release a new minor release every three months, and keep three minor releases "supported" at a time.  So we'll "deprecate" (not clear what that means) 1.0 when 1.2 comes out, and no longer support it when 1.3 arrives.

**Comment by ihmccreery** [comment]
(See [versioning.md](https://github.com/kubernetes/kubernetes/blob/master/docs/design/versioning.md)).

**Comment by wonderfly** [comment]
Thanks. Well explained.

**Pull request merged**